### PR TITLE
User can sign up with OAuth

### PR DESCRIPTION
**Pull request by piavW**
[PT-link](https://www.pivotaltracker.com/story/show/168982423)

## PR content
This PR enables the visitor to log in using Facebook authentication.

## This PR contains
- Facebook-login integration
- Feature testing where Facebook-login functionality is stubbed out
- Heroku set-up

## What we've learned
- We can manually log in to Facebook using localhost:3000, but we can't test with it since we add the facebook-login functionality in javascript (using Facebook SDK). Then Facebook has restrictions abut http requests.
- Stub out a facebook-login-function.

Screenshot 2019-10-09 at 16 47 31

### CODE REVIEW

**Review by AlexSaxena — APPROVED** [review]

**Review by jonas-masharqa — APPROVED** [review]

**Review by beccaburns — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -9,4 +9,9 @@
 /yarn-error.log
 /public/assets
 .byebug_history
-/config/master.key
\ No newline at end of file
+/config/master.key
+# Ignore master key for decrypting credentials and more.
+/config/master.key
+
```
Comment (faraznaeem) [inline .gitignore]: A lot of ignoring masterkeys in this file

File: app/controllers/recipes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  def index
+    @recipes = FoodService.get_recipes(params[:search])
+  end
+
+  private
+
+  def recipe_params
+    params.require(:recipe).permit(:search)
+  end
+end
```
Comment (faraznaeem) [inline app/controllers/recipes_controller.rb]: Does this not belong to another PR?

**Review by faraznaeem — CHANGES_REQUESTED** [review]
This PR contains changes from another PR?

File: app/controllers/recipes_controller.rb (hunk 2 above)
Comment (piavW) [reply app/controllers/recipes_controller.rb]: No it's our landing-page.  :) We need to have it.

**Review by oliverochman — APPROVED** [review]
